### PR TITLE
1478-accessibility_508_compliance

### PR DESCRIPTION
**Pull request by Dmac26**
﻿## Summary
Addresses Issue # 1478
https://app.zenhub.com/workspaces/open-forest-58f8bbe105a35fad2d275a0c/issues/usdaforestservice/fs-open-forest-platform/1478

This PR addresses the 508 compliance issues uncovered with AMP manual testing.

## This pull request is ready to merge when...
- [x] Feature branch is starts with the issue number
- [x] Is connected to its original issue via zenhub 👇
- [x] All tests are passing and meet coverage, linting, and accessibility requirements. And no security vulnerabilities ⚫️(Circle)
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

File: frontend/src/app/applications/permit-application-view/temp-outfitter-details.component.html [hunk 1]
```diff
@@ -48,7 +48,7 @@ <h2 class="app-item-label padding-right-1">Applicant information</h2>
         <span *ngIf="application.applicantInfo.fax.extension"> ext. {{ application.applicantInfo.fax.extension }}</span>
       </span>
       <span class="display: block" *ngIf="application.applicantInfo.website">
-        Website: <a href="{{application.applicantInfo.website | UrlPipe }}" target="_blank">{{application.applicantInfo.website}}</a>
+        Website: <a title="WARNING: This link will open a new window or tab to an applicant-provided URL" href="{{application.applicantInfo.website | UrlPipe }}" target="_blank">{{application.applicantInfo.website}}</a>
```
Comment (mgwalker) [inline frontend/src/app/applications/permit-application-view/temp-outfitter-details.component.html:51]: Tagging @jstrothman on this one. I think this is an anti-pattern because it's providing more information to screen reader users than to non-screen reader users. If this warning is relevant, it should be made available to everyone. I think there is a valid use of `title` here to provide extra context, but maybe it should be something like "visit the URL provided by the applicant"?

**Review by mtlaney — APPROVED** [review]

**Review by mgwalker — COMMENTED** [review]
I have one accessibility question I'd like @jstrothman to weigh in on before pushing this up.

File: frontend/src/app/applications/permit-application-view/temp-outfitter-details.component.html (hunk 1 above)
Comment (jstrothman) [reply frontend/src/app/applications/permit-application-view/temp-outfitter-details.component.html:51]: Agreed that if the information is important, the `title` attribute is problematic. Some screenreaders ignore the `title` attribute completely, e.g. VoiceOver, and the `title` attritbute is unavailable to keyboard-only users. 
Recommendation: make text visible in the screen, in parentheses following the link: "(URL provided by applicant will open in new window)"

**Review by mgwalker — CHANGES_REQUESTED** [review]
Let's take out the [link title](https://github.com/USDAForestService/fs-open-forest-platform/pull/1487#discussion_r505711016) and put that text into the link body.

File: frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html [hunk 2]
```diff
@@ -8,7 +8,7 @@ <h1 id="skip-nav-target" tabindex="-1" class="usa-heading">Firewood permit confi
 
     <div class="confirmation-details-container margin-top-2">
       <p>Your firewood permit has been purchased.</p>
-      <p>Your permit, load tags and required documents have been sent to <strong>email@address.com</strong>. If for some reason this email address is incorrect, please print your documents now.</p>
```
Comment (briandavidson) [inline frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html:11]: There was a plan to clean this up in a follow up issue after we verify we are getting the `permit` object correctly on staging. It's not that it would be very complicated to merge this change into my branch when the time comes, but for the sake of scope creep can we please omit this change as well?

File: frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html [hunk 3]
```diff
@@ -19,18 +19,16 @@ <h2>Required documents</h2>
...
         Print now
       </button>
-      <p>These documents have also been emailed to <strong>email@address.com</strong></p>
+      <p>These documents have also been emailed to <strong>{{permit.emailAddress}}</strong></p>
     </div>
 
     <div *ngIf="forest" id='tree-order-summary-container' class="confirmation-details-container clear margin-top-2">
       <div *ngIf="isPermitExpired" class="usa-alert usa-alert-error permit-expiration-warning">
         <div class="usa-alert-body">
-          <!-- <h3 class="usa-alert-heading">This permit was issued on {{permit.transactionDate | date: 'shortDate'}} and expired on {{permit.expirationDate | date: 'shortDate'}}</h3> -->
```
Comment (briandavidson) [inline frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html:33]: Same as the prior comments about this being addressed in another issue. Please omit all the changes in this block that are out of scope for 1478.

File: frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html [hunk 4]
```diff
@@ -54,40 +51,40 @@ <h2 class="usa-heading">Order summary</h2>
...
+          <td>${{permit.totalCost}}</td>
         </tr>
         <tr class="firewood-order-detail">
           <td><strong>Transaction method:</strong></td>
           <td>Credit card</td>
         </tr>
         <tr class="firewood-order-detail">
           <td><strong>Transaction date:</strong></td>
-          <td>1/2/30</td>
-          <!-- <td>{{permit.transactionDate | date: 'shortDate'}}</td> -->
```
Comment (briandavidson) [inline frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html:79]: All of these changes are out of scope as well and will be addressed after I verify results on staging. Please omit.

File: frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html [hunk 5]
```diff
@@ -44,8 +42,7 @@ <h2 class="usa-heading">Order summary</h2>
       <div class="firewood-order-detail usa-width-one-third">
         <dl>
           <dt><strong>Quantity</strong></dt>
-          <dd>4</dd>
-          <!-- <dd>{{permit.quantity}}</dd> -->
```
Comment (briandavidson) [inline frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html:48]: Out of scope. Please omit.

File: frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html [hunk 6]
```diff
@@ -54,40 +51,40 @@ <h2 class="usa-heading">Order summary</h2>
...
 
     <div id="tree-additional-info-container" class="confirmation-details-container usa-width-one-whole margin-top-2 margin-bottom-2">
       <h2 class="usa-heading">Additional information</h2>
       <hr class="header-divider">
-      <p>Your permit, load tag and required documents were emailed to you at <strong>email@address.com.</strong></p>
-      <!-- <h3>Your permit and the Christmas tree cutting rules and guidelines were emailed to you at {{permit.emailAddress}}.</h3> -->
+      <p>Your permit, load tag and required documents were emailed to you at <strong>{{permit.emailAddress}}</strong></p>
       <p>Return to the <a [routerLink]="'/firewood/forests/' + forest.forestAbbr">{{forest.forestName}} Firewood permit website.</a></p>
-      <!-- <p *ngIf="forest">Return to the <a [routerLink]="'/christmas-trees/forests/' + forest.forestAbbr">{{forest.forestName}} Christmas tree permit website.</a></p> -->
     </div>
```
Comment (briandavidson) [inline frontend/src/app/firewood/forests/order-confirmation/order-confirmation.component.html:88]: Out of scope, please omit.

File: frontend/src/app/main-landing/main-landing.component.html [hunk 7]
```diff
@@ -44,32 +44,34 @@ <h3 class="card-title">Temporary Outfitting and Guiding Permit</h3>
...
 
-  <div class="help-pick-flex" id="help-pick-background">
-    <div class="small-col small-col-left">
-      <span class="hero-icon">
-        <i class="fas fa-tree usa-ico"></i>
-      </span>
-    </div>
-    <div class="big-col">
-      <h2 class="left-align">Looking for Christmas trees?</h2>
-      <p class="left-align">Please click on the following link to be redirected to the new online sales of Christmas Tree permits for the USFS at <a href="https://www.recreation.gov/">recreation.gov</a></p>
```
Comment (briandavidson) [inline frontend/src/app/main-landing/main-landing.component.html:70]: Not sure if this block is scope creep or not. I don't have any work in progress on code, but it doesn't seem related to 508.

File: frontend/src/sass/pages/main-landing.scss [hunk 8]
```diff
@@ -116,7 +116,7 @@
   }
 }
 
-#help-pick-background{
```
Comment (briandavidson) [inline frontend/src/sass/pages/main-landing.scss:119]: small issue but if you want to address it for code cleanliness / consistency there should be a space before the `{`

File: frontend/src/sass/pages/main-landing.scss [hunk 9]
```diff
@@ -183,7 +183,7 @@
     }  
   }
 
-  #help-pick-background{
```
Comment (briandavidson) [inline frontend/src/sass/pages/main-landing.scss:186]: small issue but if you want to address it for code cleanliness / consistency there should be a space before the `{`

File: frontend/src/sass/pages/main-landing.scss [hunk 10]
```diff
@@ -183,7 +183,7 @@
...
+  #help-me-pick-background{
     padding: 50px 150px;
     .small-col-left{
```
Comment (briandavidson) [inline frontend/src/sass/pages/main-landing.scss:188]: small issue but if you want to address it for code cleanliness / consistency there should be a space before the `{`

File: server/dba/migrations/49-addPurchaseDateToFirewoodPermits.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.addColumn(TABLE_NAME, 'purchase_date', { type: Sequelize.DATE, allowNull: true }).then(() => {
+      queryInterface.sequelize.query('UPDATE "firewoodPermits" SET purchase_date = updated')
+    });
+  },
+  down: function(queryInterface, Sequelize) {
+    return queryInterface.removeColumn(TABLE_NAME, 'purchase_date');
+  }
+};
```
Comment (briandavidson) [inline server/dba/migrations/49-addPurchaseDateToFirewoodPermits.js:14]: None of this looks related to 508 stuff 😬  Should these changes be in this PR?

File: server/environment-variables/local.json [hunk 12]
```diff
@@ -55,7 +55,8 @@
       "security": [
         "bdavidson@cynerge.com"
       ],
-      "smtp_server": "smtp.sendgrid.net",
+      "port": 25,
+      "smtp_server": "mailproxy1.usda.gov",
```
Comment (briandavidson) [inline server/environment-variables/local.json:59]: I think I see what you were saying now about some stuff being pulled in from another branch @Dmac26! This looks out of scope too.

File: server/src/controllers/forests/firewood-permits.es6 [hunk 13]
```diff
@@ -67,7 +67,7 @@ firewoodPermits.getOnePermit = async (req, res) => {
   };
 
   try {
-    const permit = await firewoodModel.findOne(query);
```
Comment (briandavidson) [inline server/src/controllers/forests/firewood-permits.es6:70]: Out of scope I believe

File: server/src/controllers/forests/firewood-permits.es6 [hunk 14]
```diff
@@ -84,16 +84,11 @@ firewoodPermits.getOnePermit = async (req, res) => {
...
+  const permit = req.body;
   try {
-    const permitResponse = await firewoodPermitService.emailPDF(permit);
-    return res.status(200).send(permitResponse);
+    return res.status(200).send(firewoodPermitService.emailPDF(permit));
   } catch (error) {
-    return res.status(400).send(formatPermitError(error));
+    return res.status(400).send(error);
   }
 };
```
Comment (briandavidson) [inline server/src/controllers/forests/firewood-permits.es6:93]: This is out of scope too.

File: server/src/email/email-util.es6 [hunk 15]
```diff
@@ -115,14 +113,14 @@ emailUtil.sendPermit = (data) => {
...
-  });
+  const emailResponse = await emailUtil.transporter.sendMail(mailOptions);
+  try {
+    fs.unlinkSync('permit.pdf');
+    console.log('successfully deleted permit.pdf');
+  } catch (err) {
+    console.log('error deleting permit.pdf');
+  }
+  return emailResponse;
 };
```
Comment (briandavidson) [inline server/src/email/email-util.es6:124]: Scope creep here too

File: server/src/services/paygov/paygov.es6 [hunk 16]
```diff
@@ -59,7 +59,13 @@ paygov.createToken = (permitId) => {
  * @param {Boolean} isCancelUrl - whether to include the cancel query
  * @return {Promise} - Promise that resolves to a success URL for payGov
  */
-paygov.returnUrl = () => `${vcapConstants.INTAKE_CLIENT_BASE_URL}`;
```
Comment (briandavidson) [inline server/src/services/paygov/paygov.es6:62]: Scope creep here too.

**Review by briandavidson — CHANGES_REQUESTED** [review]
I think you were right that potentially another branch got mixed into this because I'm seeing a lot of scope creep. Sorry for so many brief comments, I was just trying to make sure every change that needs to be omitted got marked.
